src/image_builder.py

```diff
@@ -176,16 +176,16 @@ def image_builder(buildspec):
     with concurrent.futures.ThreadPoolExecutor(max_workers=10) as executor:
         # Standard images must be built before example images
         # Example images will use standard images as base
-        standard_images = [image for image in IMAGES if "example" not in image.name.lower()]
-        example_images = [image for image in IMAGES if "example" in image.name.lower()]
+        standard_images = [image for image in IMAGES if not image.info.get('base_image_uri')]
```

Comment:
update the comments above about example images

Reply:
Frankly very confusing. What do you thing about adding a property _is_base_image / is_standard_image_ into DokerImage object?

Reply:
This was just a comment to myself to update the comments above re: example images.

Do you mean the initial code is confusing, or this addition is confusing? This addition is fairly straightforward - basically says anything that has "base_image_uri" defined in the buildspec should run after anything that doesn't have "base_image_uri" defined.

We could change `standard_images` --> `base_images` if that's more clear

Reply:
I have changed this file in the following PR: https://github.com/aws/deep-learning-containers/pull/1186 

Adapting to this script into that should not be very complicated. 

Just wanted to point out that this file has been restructured a bit.

Reply:
> basically says anything that has "base_image_uri" defined in the buildspec should run after anything that doesn't have "base_image_uri" defined.
> 

Exactly. Everything that HAS "base_image_uri" IS NOT a base image. I understand why it's like that. But it's quite confusing when you read it. Having a property will let you write _if image.is_base_image_ which is much clearer.

Reply:
sure, we can add a property